tests/integration/helpers.py

```diff
@@ -497,3 +502,230 @@ def get_unit_by_index(app_name: str, units: list, index: int):
+        temp_file.write(content)
+        temp_file.flush()
+
+        subprocess.run(
+            [
+                "kubectl",
+                "cp",
+                "-n",
+                ops_test.model.info.name,
+                "-c",
+                container_name,
+                temp_file.name,
+                f"{pod_name}:{path}",
+            ],
+            check=True,
+        )
+
+    # return_code, _, _ = await ops_test.juju(
+    #     "ssh", unit.name, "chown", f"{MYSQL_SYSTEM_USER}:root", path
+    # )
+    # assert return_code == 0
+
+
+async def read_contents_from_file_in_unit(
+    ops_test: OpsTest, unit: Unit, path: str, container_name: str = CONTAINER_NAME
+) -> str:
+    """Read contents from file in the provided unit.
+
+    Args:
+        ops_test: The ops test framework
+        unit: The unit in which to read file from
+        path: The path from which to read content from
+        container_name: The container where the file exists
+
+    Returns:
+        the contents of the file
+    """
+    pod_name = unit.name.replace("/", "-")
+
+    with tempfile.NamedTemporaryFile(mode="r+") as temp_file:
+        subprocess.run(
+            [
+                "kubectl",
+                "cp",
+                "-n",
+                ops_test.model.info.name,
+                "-c",
+                container_name,
+                f"{pod_name}:{path}",
+                temp_file.name,
+            ],
+            check=True,
+        )
+
+        temp_file.seek(0)
+
+        contents = ""
+        for line in temp_file:
+            contents += line
+            contents += "\n"
+
+    return contents
+
+
+async def ls_la_in_unit(
+    ops_test: OpsTest, unit_name: str, directory: str, container_name: str = CONTAINER_NAME
+) -> list[str]:
+    """Returns the output of ls -la in unit.
+
+    Args:
+        ops_test: The ops test framework
+        unit_name: The name of unit in which to run ls -la
+        path: The path from which to run ls -la
+        container_name: The container where to run ls -la
+
+    Returns:
+        a list of files returned by ls -la
+    """
+    return_code, output, _ = await ops_test.juju(
+        "ssh", "--container", container_name, unit_name, "ls", "-la", directory
+    )
+    assert return_code == 0
+
+    ls_output = output.split("\n")[1:]
+
+    return [
+        line.strip("\r")
+        for line in ls_output
+        if len(line.strip()) > 0 and line.split()[-1] not in [".", ".."]
+    ]
+
+
+async def stop_running_log_rotate_dispatcher(ops_test: OpsTest, unit_name: str):
+    """Stop running the log rotate dispatcher script.
+
+    Args:
+        ops_test: The ops test object passed into every test case
+        unit_name: The name of the unit to be tested
+    """
+    # send TERM signal to mysql daemon, which trigger shutdown process
+    await ops_test.juju(
+        "ssh",
+        unit_name,
+        "pkill",
+        "-15",
```

Comment:
I guess `15` will work, but would advice `-9|sigkill` just in case

Reply:
addressed in 8e67290e4132069a043696468aed12fb5c884de4